pandas/tseries/holiday.py

```diff
@@ -406,17 +407,17 @@ def holidays(self, start=None, end=None, return_name=False):
         start = Timestamp(start)
         end = Timestamp(end)
 
-        holidays = None
+        holidays = []
         # If we don't have a cache or the dates are outside the prior cache, we
         # get them again
         if self._cache is None or start < self._cache[0] or end > self._cache[1]:
             for rule in self.rules:
-                rule_holidays = rule.dates(start, end, return_name=True)
+                holidays.append(rule.dates(start, end, return_name=True))
 
-                if holidays is None:
-                    holidays = rule_holidays
-                else:
-                    holidays = holidays.append(rule_holidays)
```

Comment:
This implementation also has the benefit of doing a single `concat` outside the loop rather than an O(N²) append on each iteration.